src/sentry/tasks/post_process.py

```diff
@@ -684,13 +684,17 @@ def update_event_groups(event: Event, group_states: Optional[GroupStates] = None
         # deprecated event.group and event.group_id usage, kept here for backwards compatibility
         event.group, _ = get_group_with_redirect(event.group_id)
         event.group_id = event.group.id
+        # We buffer updates to last_seen, assume its at least >= the event datetime
+        event.group.last_seen = max(event.datetime, event.group.last_seen)
 
     # Re-bind Group since we're reading the Event object
     # from cache, which may contain a stale group and project
     group_states = group_states or ([{"id": event.group_id}] if event.group_id else [])
     rebound_groups = []
     for group_state in group_states:
         rebound_group = get_group_with_redirect(group_state["id"])[0]
+        # We buffer updates to last_seen, assume its at least >= the event datetime
+        rebound_group.last_seen = max(event.datetime, rebound_group.last_seen)
```

Comment:
This isn't a great idea. If we happen to save the group it could override the last seen. We avoided this before with `times_seen` with a property like https://github.com/getsentry/sentry/blob/f977c14af2c4d2f20bfe626e7af2d949edfae885/src/sentry/tasks/post_process.py#L429.

I think it'd be safer to follow this pattern